File: server/src/Models/Database/Entry.cs
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace IsThisAMood.Models.Database
{
    public class Entry
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Name { get; set; }

        public string Mood { get; set; }

        public string Rating { get; set; }

        public List<string> Activities { get; set; }
    }
}

File: app/lib/entry.dart
class Entry {
  String id;
  String name;
  String mood;
  String rating;
  List<dynamic> activities;

  Entry(this.name, this.mood, this.rating, this.activities);

  Entry.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        mood = json['mood'],
        rating = json['rating'],
        activities = json['activities'];
}
